### PR TITLE
[PM-3513] Show error SSO policy

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Show the api error when a user tries to Login with device but there is an SSO only policy enforced.

## Code changes

Catch Api exception and show internal error message.

## Screenshots

image

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by jlf0dev — APPROVED** [review]

File: src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs [hunk 1]
```diff
@@ -277,6 +278,10 @@ private async Task CheckLoginRequestStatus()
                     await HandleLoginCompleteAsync();
                 }
             }
+            catch (ApiException ex) when (ex?.Error?.StatusCode == System.Net.HttpStatusCode.BadRequest)
+            {
+                HandleException(ex, ex?.Error?.Message);
+            }
```
Comment (fedemkr) [inline src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs:284]: ⛏️ `ex` is never null and inside given that you've already checked `null` in the `when` condition, is not necessary again.
```suggestion
            catch (ApiException ex) when (ex.Error?.StatusCode == System.Net.HttpStatusCode.BadRequest)
            {
                HandleException(ex, ex.Error.Message);
            }
```

**Review by fedemkr — CHANGES_REQUESTED** [review]

File: src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs (hunk 1 above)
Comment (fedemkr) [reply src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs:284]: 🤔 @andrebispo5  Can't remember the reason, why are we not using the base `HandleException`? which already takes care of this.

Comment (andrebispo5) [reply src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs:284]: Didn't noticed this was using a private HandleException class, this must have been before we had the base method done. I'll just remove the private and let it use the base one.

File: src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs [hunk 2]
```diff
@@ -277,6 +278,10 @@ private async Task CheckLoginRequestStatus()
                     await HandleLoginCompleteAsync();
                 }
             }
+            catch (ApiException ex) when (ex.Error?.StatusCode == System.Net.HttpStatusCode.BadRequest)
+            {
+                HandleException(ex);
+            }
```
Comment (fedemkr) [inline src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs:284]: 🤔 Any `BadRequest` should handle the exception the same or only the ones that corresponds to SSO only policy enforced?

Comment (andrebispo5) [reply src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs:284]: I don't think we will have more bad requests other than this one but if the request is bad we don't want to continue to run the check over and over. I'm even leaning towards not having the StartCheck running again in general, it causes more harm that good.

Comment (fedemkr) [reply src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs:284]: Alright I'll approve then and we can revise the start check afterwards.

**Review by fedemkr — APPROVED** [review]

**Pull request merged**